examples/main.cpp

```diff
@@ -131,9 +131,9 @@ void sample2_mlp(const string& data_dir) {
 #if defined(_MSC_VER) && _MSC_VER < 1800
   // initializer-list is not supported
   int num_units[] = {28 * 28, num_hidden_units, 10};
-  auto nn         = make_mlp<tan_h>(num_units, num_units + 3);
+  auto nn         = make_mlp<tanh_layer>(num_units, num_units + 3);
```

Comment:
@edgarriba I also would like to drop backward compatibility to improve the library with ease.

But before then, there are several kinds of backward compatibilities, such as API, binary file format.
How much are they important? Will we need to announce a change in release notes?